lib/rspec/core/hooks.rb

```diff
@@ -62,7 +62,8 @@ module Hooks
       #
       # If more than one `before` is declared within any one scope, they are run
       # in the order in which they are declared. Any `around` hooks will execute
-      # later than any `before` hook regardless of scope.
+      # after `before` context hooks but before any `before` example hook
+      # regardless of scope.
```

Comment:
It's not quite clear what does "regardless of scope" applies to. We don't support `around(:context)` now, and in this sentence, there's a clear distinction for `before`'s scopes.

Reply:
It means different context blocks, e.g. its the type of hook thats important not the block scope

Reply:
It took me a while to understand even with your explanation. Maybe remove "regardless of scope" to avoid interpretation?

Reply:
Have you got a suggested replacement?

Reply:
So, from my point of view, the problem with those statements is that "scope" is ambiguous, it can refer to both the hook scope (`:context`/`:example`) and the context in which the hook is declared.

> hooks are stored in three scopes, which are run in order: `:suite`, `:context`, and `:example`

vs

> different context blocks

(I miserably fail at "suggested edits") WDYT of the following:
```diff
-      # If more than one `before` is declared within any one scope, they are run
-      # in the order in which they are declared. Any `around` hooks will execute
-      # after `before` context hooks but before any `before` example hook
-      # regardless of scope.
+      # If more than one `before` is declared within a context, they are run
+      # in the order in which they are declared. Any `around` hooks will execute
+      # after all `before` hooks with `:context` scope, but before any `before`
+      # hook with `:example` scope regardless of the place where any of those
+      # hooks are declared.
```

I've changed "any one" to just "a", but totally uncertain if this is worse/better.

Reply:
> is declared within any one scope

or does this refer to `:context`/`:example` scope? Please leave this change off if so.